core/metacling/src/TCling.cxx

```diff
@@ -3874,7 +3874,8 @@ void TCling::SetClassInfo(TClass* cl, Bool_t reload)
 
    }
 
-   TClingClassInfo* info = new TClingClassInfo(GetInterpreterImpl(), name.c_str());
+   bool instantiateTemplate = !cl->TestBit(TClass::kUnloading);
+   TClingClassInfo* info = new TClingClassInfo(GetInterpreterImpl(), name.c_str(), instantiateTemplate);
```

Comment:
```suggestion
   // FIXME: Rather than adding an option to the TClingClassInfo, we should consider combining code 
   // that is currently in the caller (like SetUnloaded) that disable AutoLoading and AutoParsing and
   // code is in the callee (disabling template instantiation) and end up with a more explicit class:
   //      TClingClassInfoReadOnly.
   TClingClassInfo* info = new TClingClassInfo(GetInterpreterImpl(), name.c_str(), instantiateTemplate);
```